engine/src/main/scala/cromwell/jobstore/JobStoreWriterActor.scala

```diff
@@ -35,17 +37,25 @@ case class JobStoreWriterActor(jsd: JobStore,
       // Filter job completions that also have a corresponding workflow completion; these would just be
       // immediately deleted anyway.
       val jobCompletions = completions.toList collect { case j: JobCompletion if !completedWorkflowIds.contains(j.key.workflowId) => j }
-      val databaseAction = jsd.writeToDatabase(workflowCompletions, jobCompletions, batchSize)
+      val jobStoreAction: Future[Unit] = jsd.writeToDatabase(workflowCompletions, jobCompletions, batchSize)
+      val workflowStoreAction: Future[List[Int]] = Future.sequence {
+        completedWorkflowIds.map(workflowStoreAccess.deleteFromStore(_)).toList
+      }
```

Comment:
Thank you for asking.

I believe there is no desire to delete all of the workflows from the store in a single transaction. This is because they are not dependent on one another in a business logic sense.

The original code (deleted line) worked the same way, the `removeWorkflowStoreEntry` method has its own `runTransaction` call inside it so there is no way (to my knowledge) that we were previously composing multiple actions into one transaction.
`engine/src/main/scala/cromwell/jobstore/SqlJobStore.scala`:
```
_ <- completedWorkflowIds traverse sqlDatabase.removeWorkflowStoreEntry
```

I considered whether transactions would be appropriate within the scope of a single workflow, such that store deletion and job completion happen atomically. This was not the case in the old code either, so I deemed my refactor sufficient in that it replicated that behavior. Can always reconsider.

Reply:
One difference between `traverse` and `Future.sequence` is the traverse is one-by-one whereas the future is, as Khalid mentioned, parallel